### DISCUSSION TITLE
Delete handler not executed for resources when the namespace is deleted

### DISCUSSION THREAD

**Discussion by mboutet** [nolar/kopf#899] [Q&A] [answered]
I have a create, update, and delete handlers for `Ingress` resources. I expect the `Ingress` resources deletion handler to be executed when the namespace is deleted. However, this does not happen when the operator is configured to watch this specific namespace. It works if the operator is configured to watch all namespaces.

In both cases, the operator uses a service account binded to the `cluster-admin` cluster role. Therefore, it is likely not a permission issue.

Of course, if I add a deletion handler for namespaces, I can handle the cascade deletion to `Ingress` resources manually. However, it has the downside of adding finalizers on all namespaces (even if the operator is configured to only watch specific namespaces).

Is this behaviour intended? If it helps, I can try to come up with a minimal example that produces the issue.

Thank you in advance.

**Comment by nolar** [comment, marked as answer]
Hello. Yes, this seems to be a bug — I have created #900 for this.
